src/main/java/de/unistuttgart/overworldbackend/service/MinigameTaskStatisticService.java

```diff
@@ -72,59 +72,33 @@ public MinigameSuccessRateStatistic getSuccessRateOfMinigame(final UUID minigame
-            throw new IllegalArgumentException("high score percentages must end with 100");
-        }
+    public List<MinigameScoreHit> getPlayerHighscoreDistributions(final UUID minigameTaskId) {
         final List<PlayerTaskStatistic> playerTaskStatistics = playerTaskStatisticRepository.findByMinigameTaskIdOrderByHighscore(
             minigameTaskId
         );
-        final List<MinigameHighscoreDistribution> highscoreDistributions = new ArrayList<>();
-        for (int i = 0; i < highscorePercentagesToUse.size() - 1; i++) {
-            final MinigameHighscoreDistribution highscoreDistribution = new MinigameHighscoreDistribution();
-            highscoreDistribution.setFromPercentage(highscorePercentagesToUse.get(i));
-            highscoreDistribution.setToPercentage(highscorePercentagesToUse.get(i + 1));
-            highscoreDistributions.add(highscoreDistribution);
-        }
+        final List<MinigameScoreHit> highscoreDistributions = new ArrayList<>();
+        highscoreDistributions.add(new MinigameScoreHit(0, 0));
+        highscoreDistributions.add(new MinigameScoreHit(100, 0));
 
-        // calculate score time borders to time spent distribution percentage
-        int currentStatisticIndex = 0;
-        for (final MinigameHighscoreDistribution highscoreDistribution : highscoreDistributions) {
-            if (currentStatisticIndex >= playerTaskStatistics.size()) {
-                break;
-            }
-            final int endIndex = (int) (
-                (highscoreDistribution.getToPercentage() / 100.0) * (playerTaskStatistics.size() - 1)
-            );
-            PlayerTaskStatistic currentStatistic = playerTaskStatistics.get(currentStatisticIndex);
-            highscoreDistribution.setFromScore(currentStatistic.getHighscore());
-            while (currentStatisticIndex <= endIndex && currentStatisticIndex < playerTaskStatistics.size()) {
-                currentStatistic = playerTaskStatistics.get(currentStatisticIndex);
-                highscoreDistribution.addCount();
-                currentStatisticIndex++;
-            }
-            highscoreDistribution.setToScore(currentStatistic.getHighscore());
-        }
+        playerTaskStatistics
+            .stream()
+            .forEach(playerTaskStatistic -> {
+                highscoreDistributions
+                    .stream()
+                    .filter(minigameScoreHit -> minigameScoreHit.getScore() == playerTaskStatistic.getHighscore())
+                    .findFirst()
+                    .ifPresentOrElse(
+                        minigameScoreHit -> minigameScoreHit.addAmount(),
+                        () -> highscoreDistributions.add(new MinigameScoreHit(playerTaskStatistic.getHighscore(), 1))
+                    );
+            });
+        highscoreDistributions.sort(Comparator.comparing(MinigameScoreHit::getScore));
         return highscoreDistributions;
```

Comment:
```suggestion
            return playerTaskStatistics
              .parallelStream()
              .collect(Collectors.groupingByConcurrent(PlayerTaskStatistic::getHighScore))
              .computeIfAbsent(0, (key) -> new MinigameScoreHit(0, 0))
              .computeIfAbsent(100, (key) -> new MinigameScoreHit(100, 0))
              .entrySet()
              .parallelStream()
              .map(entry -> new MinigameScoreHit(entry.getKey(), entry.getValue().size()))
              .sorted(Comparator.comparing(MinigameScoreHit::getScore))
              .toList();
```
I am fairly sure this compiles and does the same thing as your current approach.